app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/ImageRegionSelectionInteractionViewModel.kt

```diff
@@ -83,6 +85,13 @@ class ImageRegionSelectionInteractionViewModel private constructor(
       this@ImageRegionSelectionInteractionViewModel.writtenTranslationContext
   }.build()
 
+  override fun getRawUserAnswer(): RawUserAnswer = RawUserAnswer.newBuilder().apply {
+    if (answerText.isNotEmpty()) {
+      val answerTextString = answerText.toString()
+      imageRegionSelection = parseClickOnImage(answerTextString)
```

Comment:
Is the selected region being properly restored via this view model's init{} block? It doesn't seem to be.

Reply:
I have tried to retain state but it won't worked, can you look at my current implementation thanks.